src/c++/vdo/base/dedupe.c

```diff
@@ -750,6 +750,7 @@ static void process_update_result(struct data_vio *agent)
 	enter_histogram_sample(vdo_from_data_vio(agent)->histograms.update_histogram,
 			       jiffies - context->submission_jiffies);
 #endif /* VDO_INTERNAL */
+	agent->dedupe_context = NULL;
 	release_context(context);
```

Comment:
Arguably we should use vdo_forget(context) since that's what that sugar does. Or is that awkward here? (This applies to all three lines added.)

Reply:
Yep, good idea

Reply:
Not with the local variable, but with the struct slot.

Reply:
Actually, I think we don't want to do the store of NULL until (unless) the atomic state change succeeds and we release the context. So the place where we're previously fetching the value from the struct isn't appropriate. And re-fetching the value from the struct with the macro later so that we can use the macro to store the null seems gratuitous...

Reply:
Yeah, that was kind of the question I was asking. Okay.